Disable call to xpc_connection_set_oneshot_instance

pvollan opened the pull request:
#### af0908f14b8b34784c4fc473a1d9ba157e0698ce
<pre>
Disable call to xpc_connection_set_oneshot_instance
<a href="https://bugs.webkit.org/show_bug.cgi?id=267368">https://bugs.webkit.org/show_bug.cgi?id=267368</a>
<a href="https://rdar.apple.com/120806906">rdar://120806906</a>

Reviewed by Brent Fulgham.

Disable call to xpc_connection_set_oneshot_instance when launching WebKit processes
as extensions, since this call is not needed, then.

* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::launchProcess):
(WebKit::ProcessLauncher::finishLaunchingProcess):

Canonical link: <a href="https://commits.webkit.org/273043@main">https://commits.webkit.org/273043@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/15124dccdd9d8cb5bacfcf70b82736c5982ae9b7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34058 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12856 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36032 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36699 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30869 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15241 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9994 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/29922 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34562 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10880 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/30401 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9491 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9601 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38005 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30946 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30746 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35701 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9725 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/7645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33617 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11510 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7842 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10287 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10537 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
```diff
@@ -197,7 +202,9 @@ static void launchWithExtensionKit(ProcessLauncher& processLauncher, ProcessLaun
     uuid_generate(uuid);
 
+#if !USE(EXTENSIONKIT)
     xpc_connection_set_oneshot_instance(m_xpcConnection.get(), uuid);
```
Does this allow us to block anything in the sandbox?

pvollan replied:
That is a good point! I do not think that is the case, since this will not affect e.g. which syscalls are being used, I believe. But that could very well be incorrect, and it would be good to check this closer.

Thanks for reviewing!

brentfulgham commented on Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
```diff
@@ -176,8 +176,13 @@ static void launchWithExtensionKit(ProcessLauncher& processLauncher, ProcessLaun
+            launcher->m_xpcConnection = [process makeLibXPCConnectionError:&error];
             launcher->m_process = WTFMove(process);
+            if (!launcher->m_xpcConnection) {
+                NSLog(@"Failed to create XPC connection, error = %@", error);
```
Does it make sense to keep the `_SEExtensionProcess` if we can't get an XPC connection, and return without finishing the process launch?

pvollan replied:
That is a good point! I removed this part of the patch, since this part was already changed by another commit.

brentfulgham reviewed (approved):
Please double-check the failure case -- should we be invalidating and clearing the `_SEExtensionProcess` if we can't create an XPC connection?

pvollan committed:
Disable call to xpc_connection_set_oneshot_instance
https://bugs.webkit.org/show_bug.cgi?id=267368
rdar://120806906

Reviewed by Brent Fulgham.

Disable call to xpc_connection_set_oneshot_instance when launching WebKit processes
as extensions, since this call is not needed, then.

* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::launchProcess):
(WebKit::ProcessLauncher::finishLaunchingProcess):

Canonical link: https://commits.webkit.org/273043@main
